hi-ml/testhiml/testhiml/test_attentionlayers.py

```diff
@@ -59,8 +61,18 @@ def test_max_pooling(dim_in: int, batch_size: int,) -> None:
 @pytest.mark.parametrize("num_heads", [1, 2])
 @pytest.mark.parametrize("dim_in", [4, 8])   # dim_in % num_heads must be 0
 @pytest.mark.parametrize("batch_size", [1, 7])
-def test_transformer_pooling(num_layers: int, num_heads: int, dim_in: int, batch_size: int) -> None:
+@pytest.mark.parametrize("transformer_dropout", [0.0, 0.3])
```

Comment:
We now have 64 combinations of inputs for this test. Do you think it might make sense to test the dropout options in a separate test function, keeping this one focused on compatibility of tensor shapes?

Reply:
+1 on avoiding combinatorial explosion

Reply:
We currently have one test function for each type of pooling layer, so I don't think we should have another function only `transformer_dropout` which is being used in 2/6 layers. To prevent too many combinations, I have fixed `transformer_dropout` parameter to one value for now. I think we could test it without parametrizing. What do you think?

Reply:
It should be fine to hard-code the dropout rate, as we don't expect any substantial behavioural changes.